contracts/helpers/TransferHelper.sol

```diff
@@ -87,20 +99,59 @@ contract TransferHelper is TransferHelperInterface, TokenTransferrer {
                     if (item.itemType == ConduitItemType.NATIVE) {
                         revert InvalidItemType();
                     } else if (item.itemType == ConduitItemType.ERC20) {
+                        // Ensure that the identifier for an ERC20 token is 0.
+                        if (item.identifier != 0) {
+                            revert InvalidERC20Identifier();
+                        }
+
+                        // Transfer ERC20 token.
                         _performERC20Transfer(
                             item.token,
                             msg.sender,
                             recipient,
                             item.amount
                         );
                     } else if (item.itemType == ConduitItemType.ERC721) {
+                        // If recipient is a contract, ensure it can receive
+                        // ERC721 tokens.
+                        if (recipient.code.length != 0) {
```

Comment:
Might only need to perform this check once, especially as the check occurs out-of-band, since there’s only a single recipient